python/pyspark/pandas/generic.py

```diff
@@ -889,6 +889,7 @@ def to_json(
         lines: bool = True,
         partition_cols: Optional[Union[str, List[str]]] = None,
         index_col: Optional[Union[str, List[str]]] = None,
+        ignoreNullFields: bool = False,
```

Comment:
If we go with the additional parameter, shall we use the keyword-only parameter as @zero323 suggested at https://github.com/apache/spark/pull/35296#discussion_r790685881, but with a snake case parameter name to be more pythonic?

```py
def to_json(
    ...,
    *,
    ignore_null_fields: bool = False,
    **options) -> ...
```

Then use the original name when merging to `options`.

Reply:
maybe let's just don't add it as a parameter for now ..... I prefer second way in https://github.com/apache/spark/pull/35296#discussion_r790685881 because technically `ignoreNullFields` is PySpark options to bypass, and makes less sense to document in pandas api doc surface .. we can just say that these `options` are for PySpark I/O options.

Reply:
If we plan to have more Pandas-specific defaults, we could also merge options with defaults dict:


```python
    def to_json(
        self,
        ...,
        **options: Any,
    ) -> Optional[str]:
       ...
       default_options: Dict[str, Any] = {"ignoreNullFields": False}
       options = {**default_options, **options}
```